### PR TITLE
INI: Fix ini for Python 3

### PR DESCRIPTION
**Pull request by nijel**
There is no official support in iniparse upstream, but for example
Debian ships patched iniparse version to work with Python 3.

### CODE REVIEW

**Comment by nijel** [comment]
Note that this actually doesn't test ini support on Python 3 in the CI as there is no official package for iniparse supporting Python 3 in the PyPI...

**Comment by unho** [comment]
The ideal solution would be to get that Debian patched version to be published in PyPI and then actually use it. @llimeht or @nijel do you know where that Debian patched version of `iniparse` is?

Anyway the PR is lgtm.

**Comment by llimeht** [comment]
The iniparse patch started at https://code.google.com/p/iniparse/issues/detail?id=22 and was incorporated into the Debian package at https://anonscm.debian.org/cgit/collab-maint/python-iniparse.git/commit/?id=fef9b4532a1a6a03aab82711f52ebde3f2d131d7 (there's also one additional patch following that one to disable some broken tests).

I found one possible new upstream for a Python 3 fork (but that module structure looks wrong and there doesn't appear to be any activity really so I don't think that's viable https://github.com/zeyosinc/iniparse).

Basically, iniparse is dead upstream no new upstream has appeared and every distro has adopted more or less the same patch as a workaround for that.

**Comment by unho** [comment]
So we just have to create a new github repo to have `iniparse` with that patch (so people can refer to it since upstream looks dead), and then create a package and publish it on PyPI. Once that is done we can update TTK requirements to use this new PyPI package and test whether it works in both Python 2 and 3.

@llimeht But before going for that I would like confirmation that the patch actually provides support for both Python 2 and Python 3.

**Comment by nijel** [comment]
Thanks for review, I've merged the changes...

**Comment by llimeht** [comment]
@unho, the test suite passes for both Python 2 and Python 3 using this patched version of iniparse: https://ci.debian.net/data/autopkgtest/unstable/amd64/t/translate-toolkit/102970/log.gz

You can't hijack that namespace on pypi so the old Python 2-only version will continue to be called iniparse there and you'd have to have a new name for it on pypi. I guess you should also consider whether you really want to become upstream for file format modules...

**Comment by nijel** [comment]
Alternatively there is also [inifile](https://pypi.python.org/pypi/inifile) library, though I haven't checked how does it compare (I've noticed it because of ITP for Debian).

**Comment by unho** [comment]
I've found this other option https://pypi.python.org/pypi/fileconfig When I have time I will investigate which way to go.
